Tasks/Common/packaging-common/nuget/NuGetToolGetter.ts

```diff
@@ -96,9 +99,33 @@ export async function getNuGet(versionSpec: string, checkLatest?: boolean, addNu
     let fullNuGetPath: string = path.join(toolPath, NUGET_EXE_FILENAME);
     taskLib.setVariable(NUGET_EXE_TOOL_PATH_ENV_VAR, fullNuGetPath);
 
+    // create a nuget posix script for nuget exe in non-windows agents
+    if (os.platform() !== "win32") {
+        generateNugetScript(toolPath, fullNuGetPath);
+    }
+    
     return fullNuGetPath;
 }
 
+function generateNugetScript(nugetToolPath: string, nugetExePath: string) {
+    var nugetScriptPath = path.join(nugetToolPath, "nuget");
+    taskLib.debug(`create nugetScriptPath ${nugetScriptPath}`);
+
+    fs.writeFile(
```

Comment:
What happens if the script is already in this path? Can you try running two instances of the NugetInstallTool back to back?

Reply:
This works fine. In the comment of fs.writeFile is says
`* Asynchronously writes data to a file, replacing the file if it already exists.`

Reply:
added fs.exist to prevent overwriting, can't overwrite due to 500 permissions which only allows read and execute